pandas/_libs/internals.pyx

```diff
@@ -639,14 +638,19 @@ def _unpickle_block(values, placement, ndim):
 
 
 @cython.freelist(64)
-cdef class SharedBlock:
+cdef class Block:
     """
     Defining __init__ in a cython class significantly improves performance.
     """
     cdef:
         public BlockPlacement _mgr_locs
         public BlockValuesRefs refs
         readonly int ndim
+        # 2023-08-15 no apparent performance improvement from declaring values
```

Comment:
I'm guessing this would make a difference if we actually had to use attributes of the ndarray that would be access via C instead of via Python, but yea that doesn't look to actually occur in the classes you've removed